File: lib/theme/fk_category_palette.dart
import 'package:flutter/material.dart';

/// FreshKeeper 食材分类语义色板。
///
/// 每个分类提供 `tint`(avatar 背景 / 软底)与 `ink`(描边 / 文字 / icon)。
/// 命名与设计稿 `data.jsx::FK_CATEGORIES` 一致(veg / fruit / meat / sea / dairy /
/// drink / sauce / grain / snack),便于查阅。
class FkCatColors {
  final Color tint;
  final Color ink;
  const FkCatColors(this.tint, this.ink);
}

class FkCategoryPalette {
  FkCategoryPalette._();

  static const veg = FkCatColors(Color(0xFFE8F3E1), Color(0xFF4F7A3A));
  static const fruit = FkCatColors(Color(0xFFFBE0D7), Color(0xFFB5523A));
  static const meat = FkCatColors(Color(0xFFFDD6CE), Color(0xFFA8442C));
  static const sea = FkCatColors(Color(0xFFD6EBF2), Color(0xFF3F7691));
  static const dairy = FkCatColors(Color(0xFFE5ECFA), Color(0xFF3F60B5));
  static const drink = FkCatColors(Color(0xFFE2EAF5), Color(0xFF4A5E91));
  static const sauce = FkCatColors(Color(0xFFF0EBE3), Color(0xFF7A6748));
  static const grain = FkCatColors(Color(0xFFFFF3D6), Color(0xFF9B7A2A));
  static const snack = FkCatColors(Color(0xFFFBE3CE), Color(0xFFA85F2C));

  static const Map<String, FkCatColors> all = {
    'veg': veg,
    'fruit': fruit,
    'meat': meat,
    'sea': sea,
    'dairy': dairy,
    'drink': drink,
    'sauce': sauce,
    'grain': grain,
    'snack': snack,
  };

  static FkCatColors of(String catId) => all[catId] ?? grain;

  /// 分类中文名(与设计稿 `data.jsx::FK_CATEGORIES.name` 一致)。
  static const Map<String, String> names = {
    'veg': '蔬菜',
    'fruit': '水果',
    'meat': '肉类',
    'sea': '海鲜',
    'dairy': '乳制品',
    'drink': '饮料',
    'sauce': '调味品',
    'grain': '主食',
    'snack': '零食',
  };
}

File: apps/ios/FreshPantry/DesignSystem/FkCategoryPalette.swift
import SwiftUI

/// Per-category color pair: `tint` (avatar background / soft fill) and `ink`
/// (stroke / text / icon). Light values ported from Flutter `FkCategoryPalette`
/// (ids match `data.jsx::FK_CATEGORIES`); dark values follow the design-system
/// soft+ink rule — deep-toned tint, brightened ink.
struct FkCategoryColors: Sendable {
    let tint: Color
    let ink: Color
}

enum FkCategoryPalette {
    static let veg = FkCategoryColors(
        tint: Color(light: 0xE8F3E1, dark: 0x243321),
        ink: Color(light: 0x4F7A3A, dark: 0x9DC785)
    )
    static let fruit = FkCategoryColors(
        tint: Color(light: 0xFBE0D7, dark: 0x3E2620),
        ink: Color(light: 0xB5523A, dark: 0xE99C84)
    )
    static let meat = FkCategoryColors(
        tint: Color(light: 0xFDD6CE, dark: 0x42231C),
        ink: Color(light: 0xA8442C, dark: 0xEF9379)
    )
    static let sea = FkCategoryColors(
        tint: Color(light: 0xD6EBF2, dark: 0x1E3038),
        ink: Color(light: 0x3F7691, dark: 0x8CC1D9)
    )
    static let dairy = FkCategoryColors(
        tint: Color(light: 0xE5ECFA, dark: 0x252F47),
        ink: Color(light: 0x3F60B5, dark: 0x9FB5E9)
    )
    static let drink = FkCategoryColors(
        tint: Color(light: 0xE2EAF5, dark: 0x242B3C),
        ink: Color(light: 0x4A5E91, dark: 0xA4B5DD)
    )
    static let sauce = FkCategoryColors(
        tint: Color(light: 0xF0EBE3, dark: 0x332C23),
        ink: Color(light: 0x7A6748, dark: 0xC9AF8C)
    )
    static let grain = FkCategoryColors(
        tint: Color(light: 0xFFF3D6, dark: 0x3D331D),
        ink: Color(light: 0x9B7A2A, dark: 0xE0C375)
    )
    static let snack = FkCategoryColors(
        tint: Color(light: 0xFBE3CE, dark: 0x3C2A1C),
        ink: Color(light: 0xA85F2C, dark: 0xE3A271)
    )

    static let all: [String: FkCategoryColors] = [
        "veg": veg, "fruit": fruit, "meat": meat, "sea": sea, "dairy": dairy,
        "drink": drink, "sauce": sauce, "grain": grain, "snack": snack,
    ]

    /// Falls back to `grain` for unknown ids (matches Flutter behavior).
    static func of(_ categoryId: String) -> FkCategoryColors {
        all[categoryId] ?? grain
    }
}
